README.rst

```diff
@@ -16,30 +16,88 @@ utility for Python.
 
--  default or :option:`--txt<gcovr --txt>`: compact human-readable summaries
--  :option:`--html<gcovr --html>`: HTML summaries
--  :option:`--html-details<gcovr --html-details>`: HTML report with annotated source files
--  :option:`-x/--xml<gcovr --xml>`: machine readable XML reports in Cobertura_ format
--  :option:`--sonarqube<gcovr --sonarqube>`: machine readable XML reports in Sonarqube format
--  :option:`--json<gcovr --json>`: JSON report with source files structure and coverage
--  :option:`--json-summary<gcovr --json-summary>`: JSON summary coverage report
--  :option:`--csv<gcovr --csv>`: CSV report summarizing the coverage of each file
--  :option:`--coveralls<gcovr --coveralls>`: machine readable JSON reports in Coveralls_ format
+.. list-table::
+   :header-rows: 1
+
+   * - CLI Option
+     - Guide
+     - Description
+
+   * - default, |abstract-option-txt|
+     - |abstract-guide-txt|
+     - compact human-readable summaries
+
+   * - |abstract-option-html|
+     - |abstract-guide-html|
+     - HTML summaries
+
+   * - |abstract-option-html-details|
+     - |abstract-guide-html-details|
+     - HTML report with annotated source files
+
+   * - |abstract-option-cobertura|
+     - |abstract-guide-cobertura|
+     - machine readable XML reports in Cobertura_ format
+
+   * - |abstract-option-sonarqube|
+     - |abstract-guide-sonarqube|
+     - machine readable XML reports in Sonarqube format
+
+   * - |abstract-option-json|
+     - |abstract-guide-json|
+     - JSON report with source file structure and coverage
+
+   * - |abstract-option-json-summary|
+     - |abstract-guide-json-summary|
+     - JSON summary coverage report
+
+   * - |abstract-option-csv|
+     - |abstract-guide-csv|
+     - CSV report summarizing the coverage of each file
+
+   * - |abstract-option-coveralls|
+     - |abstract-guide-coveralls|
+     - machine readable JSON report in Coveralls_ format
 
 Thus, gcovr can be viewed
 as a command-line alternative to the lcov_ utility, which runs gcov
 and generates an HTML-formatted report.
 The development of gcovr was motivated by the need for
 text summaries and XML reports.
 
-.. _gcov: http://gcc.gnu.org/onlinedocs/gcc/Gcov.html
-.. _coverage.py: http://nedbatchelder.com/code/coverage/
+.. _gcov: https://gcc.gnu.org/onlinedocs/gcc/Gcov.html
+.. _coverage.py: https://coverage.readthedocs.io/en/stable/
 .. _cobertura: http://cobertura.sourceforge.net/
 .. _lcov: http://ltp.sourceforge.net/coverage/lcov.php
 .. _coveralls: https://coveralls.io/
 
 .. end abstract
 
+.. The above table contains links into the documentation.
+.. Here are the default targets that are used for rendering on GH or on PyPI.
+.. The targets for within the Sphinx docs are given in doc/source/index.rst
+
+.. |abstract-option-txt| replace:: ``--txt``
+.. |abstract-option-html| replace:: ``--html``
+.. |abstract-option-html-details| replace:: ``--html-details``
+.. |abstract-option-cobertura| replace:: ``-x``, ``--xml``
```

Comment:
I think it is better to use the long option first.
```suggestion
.. |abstract-option-cobertura| replace:: ``--xml``, ``-x``
```

BTW: Can we add `--cobertura` for the `--xml` and use this in the documentation? I think `--xml` is too generic.

Reply:
Ok, I'll change to list the most preferred option first.

If you want I can sneak in a new `--cobertura` option in this PR. This was something I was also thinking about.

Reply:
Please add the option and remove the `--xml` from the overview.

Reply:
I added the `--cobertura` option. Note that this also renames the config key from `xml` to `cobertura`.

I had tried to emit a deprecation warning when the old options were used, but this turned out really difficult (details in commit message). So I just added `--cobertura` as the preferred option, without deprecating the existing `--xml` flag. However, I used this opportunity to reformat the configuration module and to add type annotations.

Reply:
If we do a incompatible change here, can we also remove the `txt-` in the config key of the option `--branch`?